src/HftApi/WebApi/OperationsController.cs

```diff
@@ -142,30 +144,32 @@ public async Task<IActionResult> GetCryptosDepositAddresses([FromRoute] string a
         /// <summary>
         /// Create Withdrawal
         /// </summary>
+        /// <param name="withdrawalId">Id of the Withdrawal</param>
         /// <param name="request"></param>
         /// <returns></returns>
         [HttpPost]
-        [Route("withdrawals")]
+        [Route("withdrawals/{withdrawalId}")]
```

Comment:
Let's remove it from the route, move to the payload,, and rename it either to request ID or idempotency ID and make it string. We can't use it directly as an ID of the operation. Id of the object should be generated by the system being invoked, not by an invoker. Also, to avoid collisions of request/idempotency ID passed by the different users, it's needed to use some user-scoped prefix for it (ID of the user can be used). Obtaining of the withdrawal ID for a given request/idempotency ID should idempotent -> there should be a persistence layer for it with atomical add-or-get semantic. In general it should look like:

```c#
var idempotencyId = $"{user.Id}-{request.IdempotencyId}";
var withdrawalId = _withdrawalIdRepository.AddOrGetBy(idempotencyId, idFactory: () => Guid.NewGuid());
```